api/desecapi/models.py

```diff
@@ -259,19 +259,17 @@ def is_covered_by_foreign_zone(self):
         assert self.name == next(iter(private_domains), self.public_suffix)
 
         # Deny registration for non-local public suffixes and for domains covered by other users' zones
-        try:
-            owner = self.owner
-        except Domain.owner.RelatedObjectDoesNotExist:
-            owner = None
-        return Domain.objects.filter(Q(name__in=private_domains) & ~Q(owner=owner)).exists()
+        return Domain.objects.filter(Q(name__in=private_domains) & ~Q(owner=self.owner_or_none)).exists()
```

Comment:
Alternative to this construction:

```
    def covers_foreign_zone(self):
        filter = Q(name__in=private_domains)
        try:
            filter &= ~Q(owner=self.owner)
        except Domain.owner.RelatedObjectDoesNotExist:
            pass
        return Domain.objects.filter(filter).exists()
```

The try-catch would have to be duplicated in `covers_foreign_zone`. Hm...